components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/defaultsequence/DefaultAuthSeqMgtServiceImpl.java

```diff
@@ -0,0 +1,613 @@
+                            isAuthenticatorIncluded.set(true);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private Map<String, Property[]> getAllLocalAuthenticators(String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        ApplicationManagementService applicationMgtService = ApplicationManagementService.getInstance();
+        return Arrays.stream(applicationMgtService
+                .getAllLocalAuthenticators(tenantDomain))
+                .collect(Collectors.toMap(LocalAuthenticatorConfig::getName, LocalAuthenticatorConfig::getProperties));
+    }
+
+    private void validateFederatedIdp(IdentityProvider idp, AtomicBoolean isAuthenticatorIncluded, List<String>
+            validationMsg, String tenantDomain) {
+
+        try {
+            IdentityProvider savedIdp = IdentityProviderManager.getInstance().getIdPByName(idp
+                    .getIdentityProviderName(), tenantDomain, false);
+            if (savedIdp.getId() == null) {
+                validationMsg.add(String.format(FEDERATED_IDP_NOT_AVAILABLE,
+                        idp.getIdentityProviderName()));
+            } else if (savedIdp.getFederatedAuthenticatorConfigs() != null) {
+                isAuthenticatorIncluded.set(true);
+                List<String> savedIdpAuthenticators = Arrays.stream(savedIdp
+                        .getFederatedAuthenticatorConfigs()).map(FederatedAuthenticatorConfig::getName)
+                        .collect(Collectors.toList());
+                for (FederatedAuthenticatorConfig federatedAuth : idp.getFederatedAuthenticatorConfigs()) {
+                    if (!savedIdpAuthenticators.contains(federatedAuth.getName())) {
+                        validationMsg.add(String.format(AUTHENTICATOR_NOT_CONFIGURED,
+                                federatedAuth.getName(), idp.getIdentityProviderName()));
+                    }
+                }
+            } else {
+                for (FederatedAuthenticatorConfig federatedAuth : idp.getFederatedAuthenticatorConfigs()) {
+                    validationMsg.add(String.format(AUTHENTICATOR_NOT_CONFIGURED,
+                            federatedAuth.getName(), idp.getIdentityProviderName()));
+                }
+            }
+        } catch (IdentityProviderManagementException e) {
+            String errorMsg = String.format(FEDERATED_IDP_NOT_AVAILABLE, idp.getIdentityProviderName());
+            log.error(errorMsg, e);
+            validationMsg.add(errorMsg);
+        }
+    }
+
+    private void checkUnsupportedXMLElements(String seqConfigXml, String tenantDomain,
+                                             String errorMsg) throws DefaultAuthSeqMgtException {
+
+        List<String> validationMsg = new ArrayList<>();
+
+        if (seqConfigXml != null) {
+            try {
+                DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
```

Comment:
There's a util that returns a secured document builder. Unless you have add specific properties to make sure it is not vulnerable to XXE or XEE attacks